src/main/webapp/components/query-manager/query-manager.tsx

```diff
@@ -64,19 +64,25 @@ const QueryCard = (props: QueryCardProps) => {
 }
 
 const QuerySelector = (props: QuerySelectorProps) => {
-  const { queries, currentQuery } = props
+  const { queries, currentQuery, onSearch } = props
   const hasQueries = queries && queries.length > 0
   const [anchorEl, handleOpen, handleClose, open] = useAnchorEl()
 
   const queryCards = queries.map(query => (
-    <QueryCard {...props} query={query} key={query.id} />
+    <QueryCard
+      {...props}
+      query={query}
+      key={query.id}
+      onSearch={() => onSearch(query.id!)}
```

Comment:
just curious, what does the syntax of `query.id!` do specifically the `!`?

Reply:
I think it assures the compiler that the optional won't be undefined. It was complaining before I added the `!` https://stackoverflow.com/a/42274019